src/SSA_stepper.jl

```diff
@@ -120,6 +124,12 @@ function DiffEqBase.solve!(integrator::SSAIntegrator)
     end
     integrator.t = end_time
 
+    # check callbacks one last time
+    if !(integrator.opts.callback.discrete_callbacks isa Tuple{})
+        DiffEqBase.apply_discrete_callback!(integrator,
+            integrator.opts.callback.discrete_callbacks...)
+    end
+
```

Comment:
@ChrisRackauckas wanted to check this one with you. As far as I can tell SSAStepper was not calling callbacks at the final time, `tspan[2]`. For the periodic callback this resulted in `affect_final = true` being ignored. I added this last callback check at the final time once time-stepping is done, but it would be good for you to check this is the right approach to handle this / consistent with other integrators. 

Note that `should_continue_solve`'s setup is to continue stepping if the next `tstop` or jump time are *less* than `end_time`, hence why the `final_affect` is not hit via `step!`.

Reply:
That's a pretty reasonable way to fix it.